File: snippets/single-file-single-thread/string-methods/java/basic.java
String text = "Hello, World!";

// Case methods
String upper = text.toUpperCase();  // "HELLO, WORLD!"
String lower = text.toLowerCase();  // "hello, world!"

// Search methods
int index = text.indexOf("World");      // 7 (-1 if not found)
boolean contains = text.contains("World"); // true
boolean starts = text.startsWith("Hello"); // true
boolean ends = text.endsWith("!");         // true

// Manipulation methods
String replaced = text.replace("World", "Java");  // "Hello, Java!"
String stripped = "  text  ".trim();       // "text"
String[] splitArray = text.split(", ");    // ["Hello", "World!"]
String joined = String.join("-", "a", "b"); // "a-b"

// Substring extraction
String sub = text.substring(0, 5);   // "Hello"

// Character methods
char c = text.charAt(0);   // 'H'
int length = text.length();  // 13

// Comparison
boolean equal = text.equals("Hello, World!");  // true
boolean equalIgnoreCase = text.equalsIgnoreCase("hello, world!");  // true

File: snippets/single-file-single-thread/string-methods/javascript/basic.js
const text = "Hello, World!";

// Case methods
const upper = text.toUpperCase();  // "HELLO, WORLD!"
const lower = text.toLowerCase();  // "hello, world!"

// Search methods
const index = text.indexOf("World");    // 7 (-1 if not found)
const contains = text.includes("World"); // true
const starts = text.startsWith("Hello"); // true
const ends = text.endsWith("!");         // true

// Manipulation methods
const replaced = text.replace("World", "JavaScript");  // "Hello, JavaScript!"
const stripped = "  text  ".trim();      // "text"
const splitArray = text.split(", ");     // ["Hello", "World!"]
const joined = ["a", "b"].join("-");     // "a-b"

// Substring extraction
const sub = text.substring(0, 5);   // "Hello"
const sliced = text.slice(7, 12);   // "World"

// Character at position
const char = text.charAt(0);   // "H"
const code = text.charCodeAt(0);  // 72
